src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -92,7 +92,31 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
             }
 
-            if (!string.IsNullOrEmpty(ApplicationUri))
+            string externalConfigPath = AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE") as string;
+            if (!string.IsNullOrEmpty(externalConfigPath))
+            {
+                if (Uri.IsWellFormedUriString(externalConfigPath, UriKind.Absolute))
+                {
+                    Uri externalConfigUri = new Uri(externalConfigPath);
+                    if (externalConfigUri.IsFile)
+                    {
+                        ApplicationConfigUri = externalConfigUri.LocalPath;
+                    }
+                }
+                else
+                {
+                    // Uri constructor would throw for relative paths but full framework doesn't.
+                    // We will mimic full framework behavior here.
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && externalConfigPath.Length >= 2 && externalConfigPath.StartsWith('\\') && externalConfigPath[1] != '\\')
+                    {
+                        // if path starts with a single backslash Path.Combine would combine with a root of BaseDirectory so we trim that character to avoid that
+                        externalConfigPath = externalConfigPath.Substring(0, externalConfigPath.Length - 1);
```

Comment:
Just wondering - does this match the .NET Framework behavior for paths starting with a single slash?

Reply:
```
using System;
using System.Reflection;

class Program
{
    static void Main(string[] args)
    {
         AppDomain.CurrentDomain.SetData("APP_CONFIG_FILE", @"\abc");
         Console.WriteLine(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile);
    }
}
```
It prints `c:\abc`, without taking BaseDirectory into account.

Reply:
To keep things simple, I would make this just:
```
if (!string.IsNullOrEmpty(externalConfigPath))
{
    if (!Path.IsPathRooted(externalConfigPath))
    {
        path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, externalConfigPath);
    }
    path = Path.GetFullPath(path);
}
else if (!string.IsNullOrEmpty(ApplicationUri))
```

Reply:
I'll go ahead with solution suggested by @JeremyKuhne (https://github.com/dotnet/runtime/pull/56748#discussion_r681991037), it's probably not covering all corner cases but it should be close enough. For the backslash starting directories I do not have any strong feeling we should support that as it's hard to describe the behavior in any docs especially since now we also work on Unix

Reply:
> I'll go ahead with solution suggested by @JeremyKuhne (https://github.com/dotnet/runtime/pull/56748#discussion_r681991037)

Is there any other existing place in dotnet/runtime that is normalizing non-fully qualified paths using this algorithm? I think it does not have very intuitive behavior, e.g. it does not match what `Path.GetFullPath(string path, string basePath)` 
 does.

>  it should be close enough

I would pick the simplest solution, among the "close enough" ones. It was the gist of my suggestion.

Reply:
Sadly `Path.IsPathRooted` will get you "relative" paths. `Path.IsPathFullyQualified` (new) is the one that gives you accurate indication of whether or not the current working directory will impact the path.

It is unlikely that customers would actually desire paths that start with `\` to be rooted to the current working directory drive. I think that is enough to be helpful here, other weird scenarios like `C:foo\` don't need addressed I think.